app/controllers/defendants_controller.rb

```diff
@@ -79,6 +82,49 @@ def error_messages
     @errors.map { |k, v| "#{k.to_s.humanize} #{v.join(', ')}" }.join("\n")
   end
 
+  def resource
+    if Feature.enabled?(:laa_references)
+      logger.info 'USING_V2_ENDPOINT'
+      CdApi::LaaReferences
+    else
+      logger.info 'USING_V1_ENDPOINT'
+      CourtDataAdaptor::Resource::LaaReference
+    end
+  end
+
+  def resource_save
+    if Feature.enabled?(:laa_references)
+      begin
+        logger.info 'CALLING_V2_MAAT_UNLINK'
+        unlink
+      rescue ActiveResource::ResourceInvalid, ActiveResource::BadRequest => e
+        logger.info 'CLIENT_ERROR_OCCURRED'
+        @laa_reference.errors.from_json(e.response.body)
+        render_edit(I18n.t('defendants.unlink.unprocessable'), @laa_reference.errors.full_messages.join(', '))
+      rescue ActiveResource::ServerError, ActiveResource::ClientError => e
+        logger.error 'SERVER_ERROR_OCCURRED'
+        log_sentry_error(e, @laa_reference.errors)
+        render_edit(I18n.t('defendants.unlink.failure'), I18n.t('error.it_helpdesk'))
+      else
+        redirect_to_edit_defendants
+      end
```

Comment:
To reduce the size of this method, maybe have two new methods:
def handle_client_error ...
def handle_server_error ...

Then call it from here?